use path instead of posix-specific path

Randelung opened the pull request:
addresses #38, but likely doesn't fully eliminate the issue.

I've noticed that there's a lot of explicit `posix` type path usage and manual path parsing etc. It seems to me that the scope of the extension is merely a bridge between karma/angular-cli and the UI, which leads me to believe that manual path wrangling (posix paths on windows and vice versa) shouldn't really be necessary.
I don't want to break stuff I don't fully understand, so I only took out any posix path manipulation that touches `angularConfigRootPath` in `angular-utils` since I can see that it's a windows path from the start, but I'm certain that the same could be done throughout the project. I generally advise against manual path manipulation and explicit platform distinction.

This change correctly identifed my angular project on my D: drive and started the angular cli (without any manual settings). Test detection and execution work as expected. I did `npm validate` and got 225/238 tests passed, the same as on `master`.

I've taken the liberty to increment the path level.

Commit:
use path instead of posix-specific path

addresses #38, but likely doesn't fully eliminate the issue.

Randelung commented:
... Actually, this probably doesn't have anything to do with #38, sorry. 😅 It addresses a different issue and I inadvertently hijacked the thread.

Randelung commented:
Oh yeah, I forgot to mention. Somebody should try it out on a Linux machine before merging.

lucono commented:
@Randelung, thanks for looking into this, and for the PR.

Karma Test Explorer project only uses posix-style paths, as they are understood by all 3 platforms (Linux, MacOS, Windows). Additionally, it doesn't use the `shell` option of `child_process.spawn` for a number of reasons (see previous issues #2, #9, and #12 for example).

Not using the `shell` option allows it to fully delegate path escaping concerns to each OS platform. Karma Test Explorer takes the same approach used by Angular [here](https://github.com/angular/angular-cli/commit/0978ff5dc826b1a861e0492dada5f173b269c648) for a lot of the same reasons.

Edit: The issue in the code is that the `resolve` should have been a `join`, and has a missing `normalizePath`. If you could address those and add unit tests (which the build will validate on all 3 platforms), I can merge it into the `beta` branch for release.

Commit:
Revert "use path instead of posix-specific path"

This reverts commit e171c52add0d96599ff19dcc5baae004bef3cde5.

Commit:
address angular file resolution

replace resolve with join where appropriate
introduce try/catch for JSON.parse
test suite for angular-utils

lucono commented:
@Randelung Thanks for the updates. I'm going to add some additional fixes and release to the store.
